astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -710,8 +712,15 @@ def test_onearg_test_ufuncs(self, ufunc):
         assert np.all(out == ufunc(q.value))
 
     def test_sign(self):
+        if sys.platform.startswith('win'):
```

Comment:
@mhvk, is this expected?

Reply:
If so, a cleaner way to catch this would be to use Numpy's ``errstate``:

```
with np.errstate(invalid='ignore'):
    ...
```

which means we don't need to both with conditionally selecting ``nullcontext``.

In general I'd use ``pytest.warns`` if it's a warning we want to see happen, and ``errstate`` or ``filterwarnings`` (as a decorator) if it's a nuisance but unrelated to the test.

Reply:
Is it unrelated though? I am not familiar enough with ufunc internals to know. And it is weird this only happens on Windows.

Reply:
I'd guess this happens even for regular ndarray, so I think we should count it as a nuisance - I think the `filterwarnings` may be the easiest - with a comment explaining that happens on windows and is really an upstream issue.

Reply:
Since @nden has identified this to be caused by Numpy bug, I retained the Windows-specific context.